### PR TITLE
Persist window state and geometry in application settings

### PR DESCRIPTION
**Pull request by ihabunek**

### CODE REVIEW

File: pugdebug/gui/main_window.py [hunk 1]
```diff
@@ -63,7 +76,8 @@ def setup_settings_window(self):
         self.central_widget_layout.addWidget(self.settings_window, 1, 0, 1, 2)
 
     def setup_toolbar(self):
-        toolbar = QToolBar()
+        toolbar = QToolBar("Main Toolbar")
+        toolbar.setObjectName("main-toolbar")
```
Comment (robertbasic) [inline pugdebug/gui/main_window.py]: Why was this needed?

Comment (ihabunek) [inline pugdebug/gui/main_window.py]: Required for saveState() on all QToolbars and QDockWidgets attached to the window.
see http://doc.qt.io/qt-5/qmainwindow.html#saveState

**Comment by robertbasic** [comment]
In QSettings docs http://doc.qt.io/qt-5/qsettings.html#restoring-the-state-of-a-gui-application it is mentioned that it's better to call `resize` and `move` than `setGeometry`. Have you looked into that?

File: pugdebug/gui/main_window.py (hunk 1 above)
Comment (robertbasic) [inline pugdebug/gui/main_window.py]: TIL. Thanks!

**Comment by ihabunek** [comment]
I didn't notice that, but I did read [the article it links to](http://doc.qt.io/qt-5/application-windows.html#window-geometry). And that article does not explain why we should not use `setGeometry`.

Dunno, I'd leave it as it is now and change if there are any problems with this code. WDYT?

**Comment by robertbasic** [comment]
Yup, works for me!
